File: Data_Structure_Topics/Binary_Search.py
# Time complexity O(logN)
# Array must be sorted 


# finding function !
def findItem (myArr,target) :
    left = 0
    right = len(myArr)
    while left<=right :
        middle = int((left+right)/2)
        if(myArr[middle]==target) :
            print(f"Item found on index : {middle}")
            return
        elif target<myArr[middle] :
            right = middle-1
        else :
            left = middle+1
    print(f"Item not found !")


# Execute start from here 

myArr = [1,4,5,6,7,9]
target = 7

findItem(myArr,target)

File: Data_Structure_Topics/Binary_Search.c
// Time complexity O(logN)
// Array must be sorted 


#include <stdio.h>
int main()
{
    int myArr [] = {2,3,5,7,8,9,19};
    int target = 5;
    int left=0;
    int right=sizeof(myArr)/sizeof(myArr[0]);
    int middle;
    
    while(left<=right){
        middle = (left+right)/2;
        if(myArr[middle]==target){
            printf("Item found at index : %d\n",middle);
            return 0;
        }else if(target>myArr[middle]){
            left = middle+1;
        }else{
            right = middle-1;
        }
    }
    printf("Item not found !");
    
    return -1;
}